docs/kubernetes-deployment.md

```diff
@@ -0,0 +1,125 @@
+
+The [deployment/](/deployment) directory contains a set of Kubernetes templates that can be used to follow the guide below.
+
+### Step 0: Retrieve git repository
+
+First step is to retrieve the repository where the k8s deployment templates are placed
+
+```bash
+$ git clone https://github.com/bitnami-labs/charts-syncer.git
+$ cd charts-syncer
+```
+
+### Step 1: Configure charts-syncer
+
+Edit the configuration file from [deployment/config/config.yaml](/deployment/config/config.yaml) and specify your source and target chart repositories. 
+You can find a reference example [here](https://github.com/bitnami-labs/charts-syncer/blob/master/charts-syncer.yaml).
+
+### Step 2 (optional): Update deployment options
+
+Edit [deployment/kustomization.yaml](/deployment/kustomization.yaml) and replace images.NewTag to point to the latest available release version. For example v0.14.0
+
+You can also change the frequency of execution of the cron job by editing the schedule property in [deployment/cronjob.yaml](/deployment/cronjob.yaml). By default, it will be run each 30 minutes.
+
+### Step 3: Configure Helm Chart/Container registries credentials
+
+If any of your Helm Chart or Container repositories require authentication
+you need to specify the credentials.
+
+The list of available credentials related keys and their meaning can be found [here](https://github.com/bitnami-labs/charts-syncer#configuration)
+
+The generation of the Kubernetes secrete can be achieved two different ways, by either
+
+#### a - Updating [deployment/config/secrets.env](/deployment/config/secrets.env)
+
+```diff
+ # Source repositories credentials
+ # Helm Chart
+-SOURCE_REPO_AUTH_USERNAME
+-SOURCE_REPO_AUTH_PASSWORD
++SOURCE_REPO_AUTH_USERNAME=my_chart_repo_username
++SOURCE_REPO_AUTH_PASSWORD=my_chart_repo_password
+ # Container images
+-SOURCE_CONTAINERS_AUTH_REGISTRY
+-SOURCE_CONTAINERS_AUTH_USERNAME
+-SOURCE_CONTAINERS_AUTH_PASSWORD
++SOURCE_CONTAINERS_AUTH_REGISTRY=container.registry.io
++SOURCE_CONTAINERS_AUTH_USERNAME=my_container_registry_username
++SOURCE_CONTAINERS_AUTH_PASSWORD=my_container_registry_password
+```
+
+#### b - Providing environment variables at deployment time
+
+You can provide the desired credentials as environment variables to the `kubectl apply -k` command
+
+```bash
+$ SOURCE_REPO_AUTH_USERNAME='my_chart_repo_username' \
+SOURCE_REPO_AUTH_PASSWORD='my_chart_repo_password' \
+SOURCE_CONTAINERS_AUTH_REGISTRY='registry.test.io' \
+SOURCE_CONTAINERS_AUTH_USERNAME='my_container_registry_username' \
+SOURCE_CONTAINERS_AUTH_PASSWORD='my_container_registry_password' \
+TARGET_REPO_AUTH_USERNAME='my_target_chart_repo_username' \
+TARGET_REPO_AUTH_PASSWORD='my_target_chart_repo_password' \
+TARGET_CONTAINERS_AUTH_USERNAME='my_target_container_registry_username' \
+TARGET_CONTAINERS_AUTH_PASSWORD='my_target_container_registry_password' \
+kubectl apply -k ./deployment
+```
+
+### 3 - Deploy the manifests to your Kubernetes cluster
```

Comment:
I think tittles are not consistent. I guess this should be instead:

```
### Step 4: Deploy the manifests to your Kubernetes cluster
```

Reply:
You are right, my headers got messed up, sorry